packages/bigquery/src/index.js

```diff
@@ -931,62 +1073,58 @@ BigQuery.prototype.query = function(options, callback) {
  * });
  */
 BigQuery.prototype.startQuery = function(options, callback) {
-  var that = this;
-
   if (is.string(options)) {
     options = {
       query: options
     };
   }
 
-  options = options || {};
-
   if (!options.query) {
     throw new Error('A SQL query string is required.');
   }
 
-  var defaults = {};
+  var query = extend(true, {
+    useLegacySql: false
+  }, options);
 
   if (options.destination) {
     if (!(options.destination instanceof Table)) {
       throw new Error('Destination must be a Table object.');
     }
-    defaults.destinationTable = {
+
+    query.destinationTable = {
       datasetId: options.destination.dataset.id,
       projectId: options.destination.dataset.bigQuery.projectId,
       tableId: options.destination.id
     };
-    delete options.destination;
+
+    delete query.destination;
   }
 
-  var body = {
+  var reqOpts = {
     configuration: {
-      query: extend(true, defaults, options)
+      query: query
     }
   };
 
-  this.request({
-    method: 'POST',
-    uri: '/jobs',
-    json: body
-  }, function(err, resp) {
-    if (err) {
-      callback(err, null, resp);
-      return;
-    }
+  if (query.dryRun) {
+    reqOpts.configuration.dryRun = query.dryRun;
+    delete query.dryRun;
+  }
 
-    var job = that.job(resp.jobReference.jobId);
-    job.metadata = resp;
+  if (query.jobPrefix) {
+    reqOpts.jobPrefix = query.jobPrefix;
+    delete query.jobPrefix;
+  }
 
-    callback(null, job, resp);
-  });
+  this.createJob(reqOpts, callback);
```

Comment:
Since it looks like `createJob` is also trying to wait for results, it seems the docs for this method are out of date.

It makes sense to me to have a callback called when the job is done, but it should not fetch all the results (in case they are very large)

Reply:
For larger queries we would recommend using `startQuery` (the lower level version) to cache the job object and either manually paginate or use the streaming version.

```js
dataset.startQuery('SELECT * FROM `table`', job => {
  job.getQueryResultsStream()
    .on('error', err => console.error(err))
    .on('data', row => console.log(row))
})
```